src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/network/PathComputationSpec.groovy

```diff
@@ -102,4 +107,74 @@ class PathComputationSpec extends HealthCheckSpecification {
         originalLatencies && originalLatencies.each { isl, latency -> database.updateIslLatency(isl, latency) }
         northbound.deleteLinkProps(northbound.getAllLinkProps())
     }
+
+    @Tidy
+    def "Target flow path computation strategy is not applied immediately in case flow was updated partially"() {
+        given: "Switch pair with two paths at least"
+        def swPair = topologyHelper.switchPairs.find { it.paths.size() >= 2 }
+
+        and: "A flow with cost strategy"
+        def latencyStrategy = PathComputationStrategy.LATENCY.toString().toLowerCase()
+        def costStrategy = PathComputationStrategy.COST.toString().toLowerCase()
+        def flow = flowHelperV2.randomFlow(swPair).tap { it.pathComputationStrategy = costStrategy }
+        flowHelperV2.addFlow(flow)
+
+        when: "Update path computation strategy(cost -> latency) via partialUpdate"
+        northbound.partialUpdate(flow.flowId, new FlowPatchDto().tap {
+            it.targetPathComputationStrategy = latencyStrategy
+        })
+        Wrappers.wait(WAIT_OFFSET / 2) { assert northbound.getFlowStatus(flow.flowId).status == FlowState.UP }
+
+        then: "Path computation strategy is not changed"
+        with(northbound.getFlow(flow.flowId)) {
+            pathComputationStrategy == costStrategy
+            targetPathComputationStrategy == latencyStrategy
+        }
+
+        and: "Flow is valid"
+        northbound.validateFlow(flow.flowId).each { direction -> assert direction.asExpected }
+
+        when: "Sync the flow"
+        northbound.synchronizeFlow(flow.flowId)
+
+        then: "Path computation strategy is updated and targetPathComputationStrategy is deleted"
+        with(northbound.getFlow(flow.flowId)) {
+            pathComputationStrategy == latencyStrategy
+            !targetPathComputationStrategy
+        }
+
```

Comment:
end of test1
start of test2

We should not overuse long scenarios. Moreover in this case scenarios a pretty much isolated and don't depend on each other, don't see any benefits in keeping them in 1 test

Reply:
done